lib/spack/spack/solver/asp.py

```diff
@@ -824,28 +834,56 @@ def _lock_prefix_from_cache_path(self, cache_path: str):
             spack.util.hash.b32_hash(cache_path), spack.util.crypto.bit_length(sys.maxsize)
         )
 
+    def _zip_cache_entry(self, cache_data: str):
+        """Compress cache entries via gzip"""
+        return gzip.compress(cache_data.encode("utf-8"))
+
+    def _unzip_cache_entry(self, cache_data: bytes):
+        """Decompresses cache entries, which are gzipped by default"""
+        return gzip.decompress(cache_data).decode("utf-8")
+
+    def _synchronize_manifest(self, old, new):
+        """Ensures that cache manifest is synchronized between the original
+        and temporary file created by the file cache write transaction manager
+
+        Returns both stream cursors to beginning of each stream
+        """
+        # synchronize manifests as old will be deleted by
+        # filecache once the context manager exits
+        # skip the manifest header data
+        old.readline()
+        cache_context = old.read()
+        old.seek(0, io.SEEK_SET)
+        # write an empty to line for manifest header
+        new.write("\n")
+        new.write(cache_context)
+        new.seek(0, io.SEEK_SET)
+
     def flush_manifest(self):
         """Updates the concretization cache manifest file after a cache write operation
         Updates the current byte count and entry counts and writes to the head of the
         manifest file"""
         manifest_file = self.root / self._cache_manifest
         manifest_file.touch(exist_ok=True)
-        with open(manifest_file, "r+", encoding="utf-8") as f:
+        with self._fc.write_transaction(self._cache_manifest) as (old, new):
```

Comment:
`write_transaction` opens a file in text mode, which makes `seek` calls questionable. It does not accept byte offsets, but rather whatever tell() returns, which is neither byte or char number; instead it's documented as an "opaque number".

So, if you continue to use `FileCache`, you should probably extend the API to accept a mode and read/write in binary. But given the other review comment, maybe FileCache isn't the best choice here given that it does not allow constant time updates to the file.

Reply:
> So, if you continue to use FileCache, you should probably extend the API to accept a file mode and read/write in binary. But given the other review comment, maybe FileCache isn't the best choice here given that it does not allow constant time updates to the file.

I think, at least you and I, we are in the binary only camp, the question just remains whether or not to use `FileCache`.